library/core/src/array/mod.rs

```diff
@@ -155,6 +155,20 @@ impl<T: fmt::Debug, const N: usize> fmt::Debug for [T; N] {
     }
 }
 
+// Note: the `#[rustc_skip_array_during_method_dispatch]` on `trait IntoIterator`
+// hides this implementation from explicit `.into_iter()` calls on editions < 2021,
+// so those calls will still resolve to the slice implementation, by reference.
+#[cfg(not(bootstrap))]
+#[stable(feature = "array_into_iter_impl", since = "1.53.0")]
```

Comment:
Would it be reasonable to add a bit of public API documentation noting that this impl doesn't apply to editions before Rust 2021? Otherwise, I think folks might assume that it does and get potentially quite confused when it doesn't.

Reply:
I do think we should add some docs, yes.

Reply:
It's more subtle than "impl doesn't apply", since everything but `.into_iter()` works normally. I'll mention the caveat here and add more details to the top-level primitive docs, where we already discuss iteration.